openapi.yaml

```diff
@@ -1,279 +1,662 @@
-swagger: '2.0'
+openapi: '3.0.2'
 info:
-  title: Express API Skeleton
-  description: Skeleton for Express APIs
-  version: v1
-  license:
-    name: GNU Affero General Public License Version 3
-    url: http://www.gnu.org/licenses/agpl-3.0.en.html
-  contact:
-    name: IS Data Architecture Team
-    url: https://is.oregonstate.edu/data-architecture
-    email: isdataarchitecture@oregonstate.edu
-schemes:
-  - https
-host: api.oregonstate.edu
-basePath: /v1
-externalDocs:
-  description: GitHub Repository
-  url: https://github.com/osu-mist/express-api-skeleton
-produces:
-  - application/json
-consumes:
-  - application/json
+  version: "2.0.0"
+  title: Locations API
+  description: Get information about OSU locations.
+servers: 
+  - url: https://api.oregonstate.edu/v2
 security:
   - OAuth2:
-      - full
+    - full
 paths:
-  /pets:
+  /locations:
     get:
-      summary: List all pets
       tags:
-        - pets
-      description: Get all pets
-      operationId: getPets
+       - locations
+      summary: Get locations
+      description: "Gets all `Location` objects. Can be filtered by parameters."
+      operationId: getLocations
       parameters:
-        - $ref: '#/parameters/pageNumber'
-        - $ref: '#/parameters/pageSize'
         - in: query
-          name: species
-          description: Search by pet species
+          name: filter[query]
```

Comment:
I do actually like the idea of having a single search field that can look across multiple attributes though. If I wanted to search for "Dixon" it's intuitive to me that I'd send `filter[search]=dixon` (using David's suggested name) whereas with `name` and `location` I wouldn't be super clear what the best on the use is. Might just be me though. Thoughts @tsoliangwu0130?

I also wonder if it would make sense to add those in addition to the `filter[search]` field so users can look for an exact or fuzzy match (depending how we implement it) for a specific field if that's what they want to do.

Reply:
Sure, I agree that it's helpful that having a search parameter to search across fields. But I also think we still need parameters for specific search. I'm open to any discussion, we can discuss about it in person or bring this up in our team meeting if you want.

Reply:
So then we'd have three filters -- `filter[search]`, `filter[location]` and `filter[name]`? That works for me.